### PR TITLE
Translation/ru ru

### PR DESCRIPTION
**Pull request by krotesk**
27.09.2023

### CODE REVIEW

**Comment by laurent22** [comment]
Some validation errors:

```
Step: Validating translations...
Error: Command failed: msgfmt -v "/home/runner/work/joplin/joplin/packages/tools/../../packages/tools/locales/ru_RU.po"
/home/runner/work/joplin/joplin/packages/tools/../../packages/tools/locales/ru_RU.po:1863: duplicate message definition...
/home/runner/work/joplin/joplin/packages/tools/../../packages/tools/locales/ru_RU.po:1860: ...this is the location of the first definition
msgfmt: found 1 fatal error
```

Please use poedit to change the file and avoid mistakes

File: packages/tools/locales/ru_RU.po [hunk 1]
```diff
@@ -4742,9 +4746,8 @@ msgid "to-do"
 msgstr "задача"
 
 #: packages/app-mobile/components/note-item.js:143
-#, fuzzy
 msgid "to-do: %s"
-msgstr "задачи: %s"
+msgstr "задание: %s"
```
Comment (roman-r-m) [inline packages/tools/locales/ru_RU.po]: "задачи" лучше, мне кажется

File: packages/tools/locales/ru_RU.po [hunk 2]
```diff
@@ -4579,20 +4580,23 @@ msgstr ""
 #: packages/app-mobile/components/ScreenHeader.tsx:278
 msgid "This note could not be deleted: %s"
 msgid_plural "These notes could not be deleted: %s"
-msgstr[0] "Эта заметка не может быть удалена: %s"
-msgstr[1] "Эти заметки не могут быть удалены: %s"
+msgstr[0] "Эту заметку не удалось удалить: %s"
+msgstr[1] "Эти заметки не удалось удалить: %s"
+msgstr[2] "Эти заметок не удалось удалить: %s"
```
Comment (roman-r-m) [inline packages/tools/locales/ru_RU.po:4585]: "эти заметок", как-то это выглядит странно, точно нужна еще одна строка? (и то же самое ниже про дублирование и перемещение)

Comment (krotesk) [reply packages/tools/locales/ru_RU.po:4585]: Poedit выдает про 0,5,6,7... понял - там должно быть "Этих заметок не удалось удалить: %s"

Comment (roman-r-m) [reply packages/tools/locales/ru_RU.po:4585]: Мне кажется, судя по коду, с таким переводом это будет выглядеть так:

> Эти заметок не удалось удалить: id1, id2, ...

Но я постараюсь сегодня вечером собрать и посмотреть, что получается.

**Comment by tessus** [comment]
@roman-r-m please let me know when everything is resolved. otherwise you can also squash and merge it when it is ready with the commit message: `All: Translation: Update ru_RU.po`

**Comment by roman-r-m** [comment]
> @roman-r-m please let me know when everything is resolved. otherwise you can also squash and merge it when it is ready with the commit message: `All: Translation: Update ru_RU.po`

Sure, leave it to me

**Comment by roman-r-m** [comment]
Having trouble applying translations for the mobile app. I can run it fine, I just don't see the changed strings.
Will continue tomorrow.

File: packages/tools/locales/ru_RU.po (hunk 2 above)
Comment (roman-r-m) [reply packages/tools/locales/ru_RU.po:4585]: It looks like multiple plurals are not supported. Looking at the code here: https://github.com/laurent22/joplin/blob/4658b97d850bd175e1fc5b2f89ba69dcfee045f0/packages/lib/locale.ts#L591-L593
you can see the number `n` is used to select singular/plural but it's not passed further to select a particular plural form.

What's also strange is looking at the files `packages/lib/locales` I don't see plural forms there at all. This makes me wonder if plurals are working at all.

**Comment by roman-r-m** [comment]
@laurent22 I'm not sure what to do about this PR.

New translation are fine except where multiple plural forms are added. This part doesn't seem to work but not because of the PR but because of how translations are resolved in Joplin.

**Comment by laurent22** [comment]
@roman-r-m, there's an issue about this here: https://github.com/laurent22/joplin/issues/8912 But indeed it seems that plural forms in general, not just for Russian, are not working at all. I see there's a "msgid_plural" field that's not currently handled by the translation script. If you have additional feedback specific to Russian, please add it to that other issue

**Comment by roman-r-m** [comment]
Thanks Laurent I didn't know this was a known issue.

Regarding this translation I have a few doubts as to how some strings are going to look in the app, but to check I need the support for plurals to work first.

We can keep it open and wait for the support to be implemented or split off the translations that are supported and merge those.

**Comment by laurent22** [comment]
@roman-r-m, I've added support for plural form in this PR: https://github.com/laurent22/joplin/pull/9033 Once CI passes I'll merge it and maybe you can give it a try with the Russian translation? To test it you'd need to add the Russian .po file and run `yarn buildTranslations` from the root.

In fact if you could suggest a test [here](https://github.com/laurent22/joplin/blob/7b5224b46c6ceeb25291a522b8b1dbfb4164c24b/packages/lib/locale.test.ts#L30), similar to the Polish one that would be great, since at the moment all tests are for singular and one plural form only (Including the Polish one since I couldn't find a good example).

**Comment by roman-r-m** [comment]
Thanks Laurent. I'll give it a try later this week or next week.

**Comment by laurent22** [comment]
Sorry I think there are probably too many conflicts now that the translations have been updated. So I'm closing this for now but feel free to create a new PR based on the latest dev version

**Pull request closed without merging**